src/test-string-view.cc

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright 2017 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gtest/gtest.h"
+
+#include <functional>
+
+#include "string-view.h"
+
+using namespace wabt;
+
+static string_view::size_type npos = string_view::npos;
+
+TEST(string_view, default_constructor) {
+  string_view sv;
+  ASSERT_EQ(nullptr, sv.data());
+  ASSERT_EQ(0U, sv.size());
+}
+
+TEST(string_view, copy_constructor) {
+  string_view sv1("foo");
+  string_view sv2(sv1);
+
+  ASSERT_EQ(3U, sv2.size());
+  ASSERT_EQ('f', sv2[0]);
```

Comment:
How about writing a helper function that takes a "const char*" and a string_view, and iterates of the strlen() of the first argument and tests if the contents of the string_view matches (via ASSERT_EQ). Then most of these tests can just call that helper function to test contents.

It will also allow you to test more strings than just "foo",